python/lsst/pipe/base/pipeline.py

```diff
@@ -143,15 +164,157 @@ def fromFile(cls, filename: str) -> Pipeline:
         Parameters
         ----------
         filename: `str`
-           A path that points to a pipeline defined in yaml format
+           A path that points to a pipeline defined in yaml format. This
+           filename may also supply additional labels to be used in subsetting
+           the loaded Pipeline. These labels are seperated from the path by a
+           collon, and may be specified as a comma seperated list, or a range
+           denoted as beginning..end. Beggining or end may be empty in which
+           case the range will be a half open interval. Unlike python iteration
+           end bounds are *INCLUDED*. Note range based selection is not well
+           defined for pipelines that are not linear in nature, and correct
+           behavior is not guaranteed, or may vary from run to run.
 
         Returns
         -------
         pipeline: `Pipeline`
+            The pipeline loaded from specified location with appropriate (if
+            any) subsetting
         """
-        pipeline = cls.fromIR(pipelineIR.PipelineIR.from_file(filename))
+        # Split up the filename and any labels that were supplied
+        filename, labelSpecifier = cls._parseFileSpecifier(filename)
+        pipeline: Pipeline = cls.fromIR(pipelineIR.PipelineIR.from_file(filename))
+
+        # If there are labels supplied, only keep those
+        if labelSpecifier is not None:
+            pipeline = pipeline.subsetFromLabels(labelSpecifier)
         return pipeline
 
+    def subsetFromLabels(self, labelSpecifier: LabelSpecifier) -> Pipeline:
+        """Subset a pipeline to contain only labels specified in labelSpecifier
+
+        Parameters
+        ----------
+        labelSpecifier : `labelSpecifier`
+            Object containing labels that describes how to subset a pipeline.
+
+        Returns
+        -------
+        pipeline : `Pipeline`
+            A new pipeline object that is a subset of the old pipeline
+
+        Raises
+        ------
+        ValueError
+            Raised if there is an issue with specified labels
+        """
+
+        pipeline = copy.deepcopy(self)
+
+        def remove_contracts(label: str):
+            """Remove any contracts that contain the given label
+
+            String comparison used in this is not the most elegant and may
+            have issues, but is the only feasible way when users can specify
+            contracts with generic strings
```

Comment:
My only issue with this is that people might not be setting various options simply because they know they are only running a subset. Consider a pipeline snippet:
```
tasks:
  modA:
    class: a.b.c
    config:
      property1: 17
  modB:
    class: a.b.d
    config:
      property2: 17
contracts:
  - modA.property1 == modB.property2
```
now A user runs this as
`pipetask ... --pipeline <path_to_pipeline>:modA -c modA.property1=19 ...`
They might be upset of confused if we validate the whole pipeline run the contracts, tell them that the contract failed when they didnt even intend to run that task.